atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/priority/NextTableToSweepProviderImpl.java

```diff
@@ -63,47 +83,126 @@ public NextTableToSweepProviderImpl(KeyValueService kvs, SweepPriorityStore swee
             Set<TableReference> allTables,
             List<SweepPriority> oldPriorities,
             Map<TableReference, SweepPriority> newPrioritiesByTableName) {
-        // Arbitrarily pick the first table alphabetically from the never-before-swept tables
-        List<TableReference> unsweptTables = Sets.difference(allTables, newPrioritiesByTableName.keySet())
-                .stream().sorted(Comparator.comparing(TableReference::getTablename)).collect(Collectors.toList());
-        if (!unsweptTables.isEmpty()) {
-            return Optional.of(unsweptTables.get(0));
-        } else {
-            double maxPriority = 0.0;
-            Optional<TableReference> toSweep = Optional.empty();
-            Collection<TableReference> toDelete = Lists.newArrayList();
-            for (SweepPriority oldPriority : oldPriorities) {
-                if (allTables.contains(oldPriority.tableRef())) {
-                    SweepPriority newPriority = newPrioritiesByTableName.get(oldPriority.tableRef());
-                    double priority = getSweepPriority(oldPriority, newPriority);
-                    if (priority > maxPriority) {
-                        maxPriority = priority;
-                        toSweep = Optional.of(oldPriority.tableRef());
-                    }
-                } else {
-                    toDelete.add(oldPriority.tableRef());
-                }
+        Map<TableReference, Double> tableToPriority = new HashMap<>(oldPriorities.size());
+
+        // If there are any unswept tables, give them highest priority.
```

Comment:
I think I prefer the previous logic — if any table unswept, sweep it. It's hard to reason about having some tables recorded in the priority table and some other tables not there, and the previous logic simplified this by just not allowing it. Defer the final decision to @tboam though.